Sources/Auth/AuthClient.swift

```diff
@@ -698,15 +698,24 @@ public final class AuthClient: Sendable {
   /// If using ``SignOutScope/others`` scope, no ``AuthChangeEvent/signedOut`` event is fired.
   /// - Parameter scope: Specifies which sessions should be logged out.
   public func signOut(scope: SignOutScope = .global) async throws {
-    do {
-      // Make sure we have a valid session.
-      _ = try await sessionManager.session()
+    let accessToken = currentSession?.accessToken
+
+    if scope != .others {
+        await sessionManager.remove()
+        eventEmitter.emit(.signedOut, session: nil)
+    }
+
+    guard let accessToken else {
+        throw AuthError.sessionNotFound
+    }
```

Comment:
Maybe we should not throw in case there isn't a session, and just log a warning and return.
That will also make sure we don't emit the signedOut event even when there is no session.

```suggestion
    guard let accessToken = currentSession?.accessToken else {
        configuration.logger?.warning("signOut called without a session")
        return
    }

    if scope != .others {
        await sessionManager.remove()
        eventEmitter.emit(.signedOut, session: nil)
    }
```

Reply:
@leoMehlig this suggestion is different than the flutter implementation.

Tagging @dshukertjr to discuss which behavior we should go with, 

1. Flutter sign out emits the signed out event even if there's no current session.
2. I'm suggesting we don't emit the event in that case

Reply:
I'm also fine to go with the suggestions. The only case this could go against expected behavior is when trying to log out all (or other sessions).

Reply:
Yeah, I like this suggestion of not emitting `signOut` event when there is no access token!

Reply:
@leoMehlig let's go with https://github.com/supabase/supabase-swift/pull/375#discussion_r1595489030 then.

Reply:
Perfect! Updated it.